Remove native_functions.yaml dependency from DistributionNormal.cu

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #67875
* __->__ #67874
* #67721
* #66980
* #66979

Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

peterbell10 committed:
Remove native_functions.yaml dependency from DistributionNormal.cu

[ghstack-poisoned]

Lezcano commented on aten/src/ATen/native/UnaryOps.h:
```diff
@@ -6,6 +6,7 @@
 
 namespace at {
 class Tensor;
```
Nit. I believe this is not needed anymore.

peterbell10 replied:
It's used by `multinomial_with_replacement_stub`.

Lezcano commented on aten/src/ATen/native/UnaryOps.h:
```diff
@@ -6,6 +6,7 @@
 namespace at {
 class Tensor;
+class TensorBase;
 struct TensorIterator;
```
Same with this line.

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"




[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"




[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

Commit:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

Commit:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

peterbell10 committed:
Rebase and fix merge conflicts on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

peterbell10 committed:
Fix overflow for 32-bit size_t on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500424).

peterbell10 committed:
Rebase on master on "Remove native_functions.yaml dependency from DistributionNormal.cu"


Differential Revision: [D32500424](https://our.internmc.facebook.com/intern/diff/D32500424)

[ghstack-poisoned]
